src/vmm/src/lib.rs

```diff
@@ -416,6 +466,13 @@ impl TryFrom<VMMConfig> for VMM {
     fn try_from(config: VMMConfig) -> Result<Self> {
         let mut vmm = VMM::new()?;
         vmm.configure_guest_memory(config.memory_config)?;
+        // Moving irqchip creation earlier here, because we use the vm_fd to register irqfds
+        // for virtio devices as part of their construction, and the register_irqfd operation
+        // fails if the irqchip hasn't been created (or so it seems).
```

Comment:
The GIC on aarch64 can only be created after the vcpus are created. With that in mind, instead of having `vmm.temp_add_devs` here, you should have it after `vmm.configure_vcpus` which will make it work on both aarch64 & x86_64 transparently.

Reply:
I've pulled the changes you made in this direction for #41. Thanks!